tokio-util/src/sync/reusable_box.rs

```diff
@@ -40,15 +40,11 @@ impl<T> ReusableBoxFuture<T> {
     where
         F: Future<Output = T> + Send + 'static,
     {
-        let layout = Layout::new::<F>();
-
-        if layout == self.layout {
-            // SAFETY: We just checked that the layout of F is correct.
-            unsafe {
-                self.set_same_layout(future);
+        match self.try_set(future) {
```

Comment:
I’d write this as:
```
if let Err(fut) = self.try_set(future) {
   *self = Self::new(fut);
}
```
😄

Reply:
I can do that. Doesn't matter much to me.